### ISSUE TITLE
Same-line WHERE clause misaligned in class declarations

### ISSUE THREAD

**Issue by ZaMan0806** [pinterest/ktlint#3380]
## Expected Behavior
When a class declaration has a `where` clause with multiple type constraints that starts on the **same line** as the class header, `standard:indent` should accept (and produce) continuation lines aligned to the column of the first type constraint after `where`, e.g.:

```kotlin
class Container<T> where T : Comparable<T>,
                         T : java.io.Serializable {
    fun body() {}
}
```

This is the same alignment style that `standard:indent` already produces/accepts correctly for **function** declarations with `where` on the same line (see the existing test `Issue 2175 - Given a function with return type and WHERE on same line`), and for both classes and functions when `where` is on its own line.

## Observed Behavior
For a class declaration with `where` on the same line, `standard:indent` rejects the correctly aligned continuation line and instead demands a fixed 4-space-multiple indent:

```
$ ktlint --log-level=debug Container.kt
Container.kt:2:1: Unexpected indentation (25) (should be 10) (standard:indent)
```

Running `ktlint -F` reformats the (already correctly aligned) input to:

```kotlin
class Container<T> where T : Comparable<T>,
          T : java.io.Serializable {
    fun body() {}
}
```

which is no longer aligned with the first constraint, and reads worse than the input.

## Steps to Reproduce

```kotlin
class Container<T> where T : Comparable<T>,
                         T : java.io.Serializable {
    fun body() {}
}
```

Run `ktlint Container.kt` -> reports `Unexpected indentation (25) (should be 10)`.

Uses default `.editorconfig` settings (`indent_size = 4`, `indent_style = space`, `ktlint_official` code style).

I traced this to `IndentationRule.kt`, function `visitClass`. When it handles the `WHERE_KEYWORD` child, it starts an indent context without specifying `childIndent`, so it always falls back to the default `indentConfig.indent` (4 spaces), regardless of whether `where` is on the same line as the class header. The equivalent function-declaration handler, `visitFun`, has an extra branch that computes the correct same-line alignment width from the column of the `where` keyword; `visitClass` is missing that branch entirely.

I confirmed via `IndentationRuleTest` that the snippet above fails with `AssertionError` when asserted with `hasNoLintViolations()`.

This looks related to (but not covered by) #3374 / #3363, which fixed `where`-clause alignment for `indent_style = tab`; this issue is about `indent_style = space` (the default), and specific to class declarations (function declarations already work correctly).

## Your Environment
* Version of ktlint used: `2.0.0-ALPHA-4` (also reproduced on latest `master`, commit `a6092d1f`)
* Relevant parts of the `.editorconfig` settings: default (`indent_size = 4`, `indent_style = space`, `ktlint_official` code style)
* Name and version (or code for custom task) of integration used: ktlint CLI (locally built shadow jar)
* Version of Gradle used (if applicable): N/A (CLI usage)
* Operating System and version: macOS
